ports/mimxrt/Makefile

```diff
@@ -484,6 +497,11 @@ $(BUILD)/firmware.bin: $(BUILD)/firmware.elf
 $(BUILD)/firmware.hex: $(BUILD)/firmware.elf
 	$(Q)$(OBJCOPY) -O ihex -R .eeprom $< $@
 
+$(BUILD)/firmware.uf2: $(BUILD)/firmware.elf
+	$(Q)$(OBJCOPY) -O binary -R .stack -R .ivt -R .flash_config $^ $@-binpart
+	$(Q)$(PYTHON) $(TOP)/tools/uf2conv.py -b $(BOOTLOADER_SIZE) -f MIMXRT10XX -c -o $@ $@-binpart
+	$(Q)rm $@-binpart
```

Comment:
This file lives in the build directory so there's no need to remove it here.

You could also create a new rule for `$(BUILD)/firmware.uf2_bin` that has the objcopy line from this rule, and then make this rule depend on that one.  Similar to how samd does it.

Also I suggest making a `UF2CONV` variable for `uf2conv.py`, same as in samd.

Reply:
OK for the non-remove. I'll look into using the variable.

Reply:
Done

Reply:
This switch only controls the creation of the .uf2 file. The memory layout is the same for boards which do not use the uf2 bootloader. That reminds me of adding the rule to add the .uf2 file to the download page. I assume that a .uf2 file is silently skipped if missing.